src/compiler/core.ts

```diff
@@ -1438,6 +1438,14 @@ namespace ts {
         }
     }
 
+    export function group<T>(values: ReadonlyArray<T>, getGroupId: (value: T) => string): T[][] {

```

Comment:
Should the return type be readonly?

Reply:
The compiler would warn us if we tried returning a ReadonlyArray with this declared type, but since we're not using the output mutably might as well make it readonly, sure.